source/client.ts

```diff
@@ -214,7 +214,10 @@ function discoverInterpreterMode(
 
   if (!cabal && stack) {
     // If the user only has Stack available, then use Stack.
-    return InterpreterMode.Stack;
+    if (stackProject) {
+      return InterpreterMode.Stack;
+    }
+    return InterpreterMode.StackNonProject;
```

Comment:
Actually I don't need to do this. It will be covered by the other check later on. 
```suggestion
    return InterpreterMode.Stack;
```